python_modules/dagster-graphql/dagster_graphql/implementation/fetch_asset_checks.py

```diff
@@ -18,62 +18,66 @@
     GrapheneAssetCheckCanExecuteIndividually,
     GrapheneAssetCheckExecution,
     GrapheneAssetCheckNeedsMigrationError,
+    GrapheneAssetCheckNeedsUserCodeUpgrade,
     GrapheneAssetChecks,
 )
 
 if TYPE_CHECKING:
     from ..schema.util import ResolveInfo
 
 
-def _fetch_asset_checks(
+def has_asset_checks(
+    graphene_info: "ResolveInfo",
+    asset_key: AssetKey,
+) -> bool:
+    for location in graphene_info.context.code_locations:
+        for repository in location.get_repositories().values():
+            for external_check in repository.external_repository_data.external_asset_checks or []:
+                if external_check.asset_key == asset_key:
+                    return True
+    return False
+
+
+def fetch_asset_checks(
     graphene_info: "ResolveInfo",
     asset_key: AssetKey,
     check_name: Optional[str] = None,
-) -> GrapheneAssetChecks:
+) -> Union[
+    GrapheneAssetCheckNeedsMigrationError,
+    GrapheneAssetCheckNeedsUserCodeUpgrade,
+    GrapheneAssetChecks,
+]:
+    if not graphene_info.context.instance.event_log_storage.supports_asset_checks:
+        return GrapheneAssetCheckNeedsMigrationError(
+            message="Asset checks require an instance migration. Run `dagster instance migrate`."
+        )
+
     external_asset_checks = []
     for location in graphene_info.context.code_locations:
         # check if the code location is too old to support executing asset checks individually
         code_location_version = (location.get_dagster_library_versions() or {}).get("dagster")
-        if code_location_version and version.parse(code_location_version) < version.parse("1.4.14"):
-            can_execute_individually = (
```

Comment:
this logic for disabling the execute button on old code isn't necessary anymore. Not removing the resolver because we'll use it with https://github.com/dagster-io/dagster/pull/16924